server/src/test/java/org/opensearch/search/query/QueryPhaseTests.java

```diff
@@ -1145,6 +1153,114 @@ public void testQueryTimeoutChecker() throws Exception {
+        );
+
+        Index index = new Index("IndexName", "UUID");
+        ShardId shardId = new ShardId(index, 0);
+        long nowInMillis = System.currentTimeMillis();
+        String clusterAlias = randomBoolean() ? null : randomAlphaOfLengthBetween(3, 10);
+        SearchRequest searchRequest = new SearchRequest();
+        searchRequest.allowPartialSearchResults(randomBoolean());
+        ShardSearchRequest request = new ShardSearchRequest(
+            OriginalIndices.NONE,
+            searchRequest,
+            shardId,
+            1,
+            AliasFilter.EMPTY,
+            1f,
+            nowInMillis,
+            clusterAlias,
+            Strings.EMPTY_ARRAY
+        );
+        TestSearchContextWithRequest searchContext = new TestSearchContextWithRequest(
+            queryShardContext,
+            indexShard,
+            newEarlyTerminationContextSearcher(reader, 0, executor),
+            request
+        );
+
+        QueryPhase queryPhase = new QueryPhase(delayedQueryPhaseSearcher);
+        queryPhase.execute(searchContext);
+        long tookTime = searchContext.queryResult().getTookTimeNanos();
+        assertTrue(tookTime >= (long) sleepMillis * 1000000);
+        reader.close();
+        dir.close();
+    }
+
+    private class TestSearchContextWithRequest extends TestSearchContext {
+        ShardSearchRequest request;
+        Query query;
+
+        public TestSearchContextWithRequest(
+            QueryShardContext queryShardContext,
+            IndexShard indexShard,
+            ContextIndexSearcher searcher,
+            ShardSearchRequest request
+        ) {
+            super(queryShardContext, indexShard, searcher);
+            this.request = request;
+            this.query = new TermQuery(new Term("foo", "bar"));
+        }
+
+        @Override
+        public ShardSearchRequest request() {
+            return request;
+        }
+
+        @Override
+        public Query query() {
+            return this.query;
+        }
+    }
+
+    private class DelayedQueryPhaseSearcher extends QueryPhase.DefaultQueryPhaseSearcher implements QueryPhaseSearcher {
+        // add delay into searchWith
+        private final int sleepMillis;
+
+        public DelayedQueryPhaseSearcher(int sleepMillis) {
+            super();
+            this.sleepMillis = sleepMillis;
+        }
+
+        @Override
+        public boolean searchWith(
+            SearchContext searchContext,
+            ContextIndexSearcher searcher,
+            Query query,
+            LinkedList<QueryCollectorContext> collectors,
+            boolean hasFilterCollector,
+            boolean hasTimeout
+        ) throws IOException {
+            try {
+                Thread.sleep(sleepMillis);
+            } catch (Exception ignored) {}
+            return super.searchWith(searchContext, searcher, query, collectors, hasFilterCollector, hasTimeout);
```

Comment:
Can be written as try finally block?

Reply:
When I put the return statement in a finally block, it creates a warning which causes ./gradlew precommit to fail.